tensorflow/core/kernels/dml_resource_scatter_ops.cc

```diff
@@ -281,7 +282,7 @@ class DmlResourceScatterNDBinaryKernel : public DmlKernel {
 
     DmlTensorInfo empty_tensor;
     empty_tensor.desc = DmlTensorDesc::Create(params_tensor.dtype(),
-                                              in_out_shape, in_out_shape);
+                                              in_out_shape, empty_scalar_shape);
 
     DmlKernelTensors tensors;
     tensors.inputs = {in_out_tensor, indices_tensor, updates_tensor,

```

Comment:
🤔 What does `empty_tensor` correspond to? SCATTER_ND only takes 3 inputs:

```
struct DML_SCATTER_ND_OPERATOR_DESC
{
    const DML_TENSOR_DESC* InputTensor;
    const DML_TENSOR_DESC* IndicesTensor;
    const DML_TENSOR_DESC* UpdatesTensor;
    const DML_TENSOR_DESC* OutputTensor;
    UINT InputDimensionCount;
    UINT IndicesDimensionCount;
};
```

Reply:
`empty_tensor` is what we feed to `DML_SCATTER_ND_OPERATOR_DESC.InputTensor`. DML doesn't support ScatterNdAdd or ScatterNbSub, so we do the scatter on an tensor full of zeros, and then add it to the original input.

```
// First, perform the scatter on an empty tensor
auto result = dml::ScatterND(empty_input, indices, updates,
                             in_out_shape.dims(), indices_shape.dims());

// Then, perform the binary operation on the scattered tensor and the
// original input tensor
result = BinaryOp()(input, result);
```